packages/sdk/contractkit/src/address-registry.ts

```diff
@@ -22,9 +22,9 @@ export class AddressRegistry {
   private readonly registry: Registry
   private readonly cache: Map<CeloContract, Address> = new Map()
 
-  constructor(kit: ContractKit) {
+  constructor(readonly connection: Connection) {
     this.cache.set(CeloContract.Registry, REGISTRY_CONTRACT_ADDRESS)
-    this.registry = newRegistry(kit.connection.web3, REGISTRY_CONTRACT_ADDRESS)
+    this.registry = newRegistry(connection.web3, REGISTRY_CONTRACT_ADDRESS)
```

Comment:
I do think this is a great idea, yet given that It would first involve updating the we3 celo typescript generator, and given how it by itself doesnt give us much, and given that the amount that has already been changed in v2 I think Its best to wait. 

I believe we could do this in steps by 

1. updating the generator template 
2. adding a `contractMaker` or `newContract` function to Connection
3. moving the generated contracts (sans web3 param) to their own package (would also make it possible to move sdk out of monorepo) 
4. to avoid breaking changes the contractkit could continue to reexport our the `new*` functions like exist now